meta/st0x-spec.md

```diff
@@ -0,0 +1,96 @@
+# st0x Server API Specification V1
+
+## Goal
+
+Provide signed price quotes for specific trading pairs that can be used by solvers in Raindex orders. Each endpoint handles a single trading pair and returns a signed context containing quote information.
+
+## Magic Numbers
+
+- Web data metadata magic number: `0xff5dcce9b571ba42`
+- st0x API magic number: `0xff5d577062616972` ("st0x pair" in hex)
+
+## Server Advertisement
+
+Servers advertise price quote endpoints on metaboard with:
+- 8 byte magic number for st0x API
+- Remaining bytes contain UTF-8 encoded URL where the API is hosted for a specific trading pair
+
+Each trading pair requires its own advertisement. Orders reference these advertisements by their hash to save gas.
+
+## Request Format
+
+Clients (solvers) POST raw octet streams to the server URL with the following binary format:
+
+```
+[ 65 byte ECDSA signature ]  # Signs the hash of all following bytes
```

Comment:
this is a copy of the same from the other spec, so i think we should reference the other spec rather than duplicate it here, which is fragile

Reply:
updated to reference the other spec